pep-9999.rst

```diff
@@ -0,0 +1,109 @@
+``bdist_egg`` distribution type on PyPI. In a parallel move, this PEP recommends
+deprecating and then disabling new uploads of distribution filenames that have
+the ``.egg`` suffix.
+
+It does not recommend removing or otherwise affecting any previously
+uploaded ``bdist_egg`` distributions.
+
+Rationale
+=========
+
+Previous Work
+-------------
+
+The groundwork for this proposal was established with :pep:`527`, which
+proposed deprecating and eventually removing upload support for a handful
+of un(der)used file extensions and distribution types.
+
+In particular, :pep:`527` proposed the removal of the ``bdist_dumb``,
+``bdist_rpm``, ``bdist_dmg``, ``bdist_msi``, and ``bdist_wininst`` distribution
+types, as well as the ``.tar``, ``.tar.bz2``, ``.tar.xz``, ``.tar.Z``,
+``.tgz``, and ``.tbz`` file extensions on distribution filenames.
+
+:pep:`527` was fully enacted with
+`PR #7529 <https://github.com/pypi/warehouse/pull/7529>`_ to Warehouse,
+which was merged on 13 April 2020.
+
+The ``bdist_egg`` format
+------------------------
+
+The ``bdist_egg`` filetype identifies distributions in the
+`Egg format <https://packaging.python.org/en/latest/glossary/#term-Egg>`_. The
+Egg format was introduced by setuptools in 2004 and is roughly equivalent
+in functionality to the
+`Wheel format <https://packaging.python.org/en/latest/glossary/#term-Wheel>`_
+introduced and standardized by :pep:`427` in 2012.
+
+Despite its longevity, the Egg format has had
+`limited adoption on PyPI <https://github.com/pypi/warehouse/issues/10653>`_.
+Some observations from that issue:
+
+* As of December 2021, ``bdist_egg`` uploads account for less than 1% of all
+  built distribution uploads;
+* ``pip`` deprecated its ``--egg`` option in 2016;
+* ``setuptools`` has considered Egg support deprecated since 2019;
+* ``build`` only supports the ``sdist`` and ``bdist_wheel`` filetypes.
+
+Given the above, this PEP proposes the removal of the ``bdist_egg`` format
+under the same justifications presented in :pep:`527`, namely:
+
+1. Egg distributions are of of limited use to the broader ecosystem and
+   therefore represent a non-reciprocal maintenance burden;
+2. Continued support for Egg distributions represents a bandwidth and storage
+   cost to PyPI, as well as to any external mirrors of PyPI.
+
+This PEP *additionally* offers an argument for removal rooted in
+standardization and duplication: the Egg format is not standardized by any
+PEP or other community standard, and overlaps heavily with its standardized
+and well-supported alternative (Wheel).
+
+The ``.egg`` file extension
+---------------------------
+
+The ``.egg`` file extension is used exclusively for distributions of the
+``bdist_egg`` format. As such, it serves no purpose in a scenario where
+PyPI disables new distribution uploads for ``bdist_egg`` distributions.
+
+Removal Process
+===============
+
+This PEP proposes a removal process based on :pep:`527`, but with the following
+modifications:
```

Comment:
Given this is essentially the equivalent of the "specification", it ends up being not _that_ much shorter than the referenced section in PEP 527, and needing to both read that (rather dense) section and then apply the multiple deltas on top of that is not the easiest to understand, IMO could you just state the plan in full here in ordered-list or bullet point form (maybe retaining a brief mention of the deltas from PEP 527, perhaps in a subsection)?

Reply:
I've replaced this whole section with a single reference to PEP 527: [6d53cac](https://github.com/python/peps/pull/3161/commits/6d53cacec66c3a28ca8e53868c4c826f04d7220b)

(Let me know if that's okay -- otherwise, I can duplicate the whole process and make a small reference as you suggested.)

Reply:
Sounds good, doing!

Reply:
Done with 113abc18: I've more or less copied, with some reflow and edits to clarify that only `sdist` and `bdist_wheel` will be allowed after this PEP.